File: llm_generated_code/Gemini/Python/Level2/Q3.py
def is_large_number(num):
    """
    Checks if a number is a large number (all digits between 5 and 9).

    Args:
        num: The number to check.

    Returns:
        True if the number is large, False otherwise.
    """

    while num > 0:
        digit = num % 10
        if digit < 5 or digit > 9:
            return False
        num //= 10
    return True

def can_be_sum_of_large_numbers(x):
    """
    Checks if a number can be expressed as the sum of two large numbers with the same number of digits.

    Args:
        x: The number to check.

    Returns:
        True if x can be expressed as the sum, False otherwise.
    """

    num_digits = len(str(x))

    # If the first digit is less than 10, it's impossible.
    if x // 10**(num_digits - 1) < 10:
        return False

    # If the last digit is greater than 9, it's impossible.
    return x % 10 <= 9

if __name__ == "__main__":
    t = int(input())

    for _ in range(t):
        x = int(input())
        if can_be_sum_of_large_numbers(x):
            print("Yes")
        else:
            print("No")

File: llm_generated_code/Gemini/CPP/Level2/Q3.cpp
#include <iostream>
#include <cmath>

using namespace std;

bool canBeSumOfLargeIntegers(int x) {
    int digits = 0;
    int temp = x;
    while (temp > 0) {
        digits++;
        temp /= 10;
    }

    // Check if the first digit is greater than or equal to 10
    // and the last digit is less than or equal to 9.
    return x / pow(10, digits - 1) >= 10 && x % 10 <= 9;
}

int main() {
    int t;
    cin >> t;

    while (t--) {
        int x;
        cin >> x;

        if (canBeSumOfLargeIntegers(x)) {
            cout << "Yes" << endl;
        } else {
            cout << "No" << endl;
        }
    }

    return 0;
}
